src/bin/languageserver/mod.rs

```diff
@@ -114,6 +134,7 @@ pub struct SolangServer {
     importmaps: Vec<(String, PathBuf)>,
     files: Mutex<Files>,
     definitions: Mutex<Definitions>,
+    types: Mutex<Types>,
```

Comment:
These two can be protected by the same mutex, right?

Reply:
yes, I have made changes so that both `definitions` and `types` use a single `Mutex`. 
I have named the field `global_cache` as it caches things required by the language server "globally" (as opposed to the per-file caches represented by the `files` field). 
Please let me know if you have another name in mind.